### ISSUE TITLE
FSharp.NET.Sdk 1.0.2 SkipCompilerExecution not supported when building with Mono's MSBuild

### ISSUE THREAD

**Issue by mrward** [dotnet/netcorecli-fsc#96]
The latest FSharp.NET.Sdk 1.0.2 NuGet package causes a build error when building with Mono's MSBuild. Building with dotnet cli works fine. FSharp.NET.Sdk 1.0.1 works fine.

# To repro (on OSX):

1. Install Mono 4.8 https://dl.xamarin.com/MonoFrameworkMDK/Macx86/MonoFramework-MDK-4.8.0.520.macos10.xamarin.universal.pkg
2. Install .NET Core SDK 1.0.1
3. Create a directory for the fsharp project.
4. dotnet new console -lang F#
5. dotnet restore
6. export MSBuildSDKsPath=/usr/local/share/dotnet/sdk/1.0.1/Sdks/
7. msbuild

# Expected result:

Build succeeds.

# Actual result:

Build fails.

```
~/.nuget/packages/fsharp.net.sdk/1.0.2/build/FSharp.NET.Core.Sdk.targets(210,15): error MSB4064: The "SkipCompilerExecution" parameter is not supported by the "Fsc" task. Verify the parameter exists on the task, and it is a settable public instance property. [~/Projects/Tests/abc.fsproj]
~/.nuget/packages/fsharp.net.sdk/1.0.2/build/FSharp.NET.Core.Sdk.targets(170,9): error MSB4063: The "Fsc" task could not be initialized with its input parameters.  [~/Projects/Tests/abc.fsproj]
```

Editing the .fsproj so the FSharp.NET.Sdk package is 1.0.1, re-running dotnet restore, then msbuild works.

**Comment by mrward** [comment]
//cc @radical

**Comment by enricosada** [comment]
@mrward with `1.0.2` the target file was changed (https://github.com/dotnet/netcorecli-fsc/commit/b14976b5cf582975d439166669d4e2eafe740db9), but the new `Fsc` task in embedded in referenced `FSharp.Compiler.Tools` 4.1.1 package.
why it's not used on mono?

sry about that, my bad, i need to add mono to ci #78 , but for sure next time i'll test manually before release a new version.

**Comment by mrward** [comment]
It looks like FSharp.Compiler.Tools 4.1.1 is being used when compiling.

```
Set Property: TargetProfile=netcore
Set Property: _OtherFscFlags=--simpleresolution --nocopyfsharpcore
Set Property: _DotNetHostExecutableName=dotnet
Set Property: _DotNetHostExecutableDirectory=/usr/local/share/dotnet/sdk/1.0.1/Sdks/../../..
Set Property: _DotNetHostExecutablePath=/usr/local/share/dotnet/sdk/1.0.1/Sdks/../../../dotnet
Set Property: _FscTask_FscToolExe=dotnet
Set Property: _FscTask_FscToolPath=/usr/local/share/dotnet/sdk/1.0.1/Sdks/../../..
Set Property: _FscTask_DotnetFscCompilerPath= "~/.nuget/packages/fsharp.compiler.tools/4.1.1/build/../tools/fsc.exe"
Using "Fsc" task from assembly "~/.nuget/packages/fsharp.compiler.tools/4.1.1/build/../tools/FSharp.Build.dll".
Task "Fsc" (TaskId:22)

...

  Task Parameter:TargetType=Exe (TaskId:22)
  Task Parameter:ToolExe=dotnet (TaskId:22)
  Task Parameter:ToolPath=/usr/local/share/dotnet/sdk/1.0.1/Sdks/../../.. (TaskId:22)
  Task Parameter:TreatWarningsAsErrors=False (TaskId:22)
  Task Parameter:WarningLevel=4 (TaskId:22)
  Task Parameter:TargetProfile=netcore (TaskId:22)
  Task Parameter:DotnetFscCompilerPath= "~/.nuget/packages/fsharp.compiler.tools/4.1.1/build/../tools/fsc.exe" (TaskId:22)
~/.nuget/packages/fsharp.net.sdk/1.0.2/build/FSharp.NET.Core.Sdk.targets(210,15): error MSB4064: The "SkipCompilerExecution" parameter is not supported by the "Fsc" task. Verify the parameter exists on the task, and it is a settable public instance property. [~/Projects/Tests/abc.fsproj]
~/.nuget/packages/fsharp.net.sdk/1.0.2/build/FSharp.NET.Core.Sdk.targets(170,9): error MSB4063: The "Fsc" task could not be initialized with its input parameters.  [/~/Projects/Tests/abc.fsproj]
Done executing task "Fsc" -- FAILED. (TaskId:22)
```

Running a dotnet build uses a different assembly for the Fsc task.

```
                   Set Property: _FscTask_DotnetFscCompilerPath= "~/.nuget/packages/fsharp.compiler.tools/4.1.1/build/netcoreapp1.0/fsc.dll"
                   Using "Fsc" task from assembly "~/.nuget/packages/fsharp.compiler.tools/4.1.1/build/netcoreapp1.0/FSharp.Build.dll".
```

Looking at the FSharp.Build.dll that is used when building with Mono and it does not seem to have a SkipCompilerExecution property.

**Comment by enricosada** [comment]
The `FSharp.Build.dll` is embedded in `FSharp.Compiler.Tools` 4.1.1 too, and that's used by .net core as the fsc task assembly.
@mrward on mono is overridden, or another is used?

**Comment by mrward** [comment]
It looks like FSharp.Build.dll from FSharp.Compiler.Tools 4.1.1 is being used from the Mono MSBuild output above. That is the assembly I was looking at which does not have a SkipCompilerExecution property. The FscToolPath and the FscTaskAssemblyPath is not being overridden by Mono as far as I can see.

```
FscTaskAssemblyPath = ~/.nuget/packages/fsharp.compiler.tools/4.1.1/build/../tools/FSharp.Build.dll
```

**Comment by enricosada** [comment]
@mrward got it, it's use .net full of `FSharp.Build.dll`, probably there was a packaing error, and is old.
instead the /build/netcoreapp1.0/FSharp.Build.dll` is ok

**Comment by enricosada** [comment]
@mrward added https://github.com/fsharp/fsharp/issues/684 to track that
We need to udpate `/tools/FSharp.Build.dll` inside `FSharp.Compiler.Tools`

**Comment by mrward** [comment]
Can we get a new FSharp.NET.Sdk NuGet package that uses a later FSharp.Compiler.Tools version? Currently NuGet picks the lowest version which is 4.1.1 when FSharp.NET.Sdk 1.0.2 is used which does not work when building with MSBuild on Mono.

I suspect just updating the dependency version is not enough though. I tried that locally and built my own FSharp.NET.Sdk 1.0.3 NuGet package using FSharp.Compiler.Tools 4.1.2 (also tried 4.1.15) but the build fails with other errors about duplicate items. Fixing the duplicate file items, then the build fails with an error about the CreateManifestResourceNames target being missing. I get these errors building with Mono's MSBuild and using dotnet build from the command line.

**Comment by enricosada** [comment]
@mrward I am testing mono 5.0 now, in a vm, before release the 1.0.3 

You didn't do `restore`

Btw `msbuild /t:Build` now work  son mono, if I remove the hack to hardcoded fsctoolpath in mono bundled fsharp.net sdk

**Comment by enricosada** [comment]
I'll release 1.0.3 in a matter of hour, but my mono 5.0 (from command line, removed the hack) seems ok for both netcoreapp and net45

**Comment by enricosada** [comment]
@mrward now f# will use

- fsc.dll if run from `dotnet build` (msbuild host is core 
- `mono fsc.exe` If run from `msbuold` 

Obv path are fsc insider nupkg

That's better ihmo, less issues to locate dotnet path and also everything will work without dotnet cli installed

**Comment by enricosada** [comment]
Published [v1.0.3](https://github.com/dotnet/netcorecli-fsc/releases/tag/v1.0.3)

The msbuild task is fixed

@mrward @nosami i tried with a clean ubuntu vm, with mono `5.0 Alpha 3 (5.0.0.61)`, and works, but you need to remove the previous workaround (the hardcoded `FscToolPath` and `FscToolExe`), more info in https://bugzilla.xamarin.com/show_bug.cgi?id=55626

I tried building `netstandard`, `net45` just with mono `msbuild` and works.
if run from mono msbuild, will use `fsc` net40, so `mono fsc.exe` from `FSharp.Compiler.Tools` 4.1.15.
This is similar to how msbuild.exe (.net frameworks) works now in windows.

For .net core, will continue to use fsc.dll (netcoreapp1.0).

I think is really ok to do it like that, because i dont need to try locate the `dotnet` if run with mono as msbuild host runtime

Going to close because the `FscTask` issue is fixed
